### PR TITLE
Adding exception case for identifiers in catch block

### PR DESCRIPTION
**Pull request by akuleshov7**
### What's done:
1) Added utility methods
2) Logic to ignore warnings for identifier e in catch block
3) Tests and code style updates

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt [hunk 1]
```diff
@@ -193,6 +193,18 @@ fun ASTNode.findAllNodesWithSpecificType(elementType: IElementType): List<ASTNod
...
 
+/**
+ * This method recursively finds parent node that has specified elementType
+ */
+fun ASTNode.findParentNodeWithSpecificType(elementType: IElementType): ASTNode? {
+    if (this.elementType == elementType) return this
+    if (this.treeParent == null) {
+        return null
+    } else {
+        return this.treeParent.findParentNodeWithSpecificType(elementType)
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt:204]: Isn't `ASTNode.parents()` from kotlin compiler which returns `sequence` better than recursion?

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt [hunk 2]
```diff
@@ -279,9 +280,21 @@ class IdentifierNaming : Rule("identifier-naming") {
...
     }
+
+    /**
+     * exception case for identifiers used in catch block:
+     * catch (e: Exception) {}
+     */
+    private fun validCatchIdentifier(node: ASTNode): Boolean {
+        return node.text == "e" &&
+                node.findParentNodeWithSpecificType(CATCH) != null &&
+                node.findParentNodeWithSpecificType(VALUE_PARAMETER_LIST) != null
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt:298]: Won't this condition also trigger on, e.g., parameters of functions declared inside catch block? I think checking type of parents with explicit index would be more reliable.

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt (hunk 1 above)
Comment (akuleshov7) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt:204]: thank you so much, didn't know about that method!

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt (hunk 2 above)
Comment (akuleshov7) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt:298]: you think that in `catch ()` block there can be something except the `e: Exception`? 
Could you please provide example just for academic interest?

Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt:298]: No, I mean something like the following:
```
CATCH: "catch (e: Exception) {
        fun foo(e: Int) { } 
    }"
- catch: "catch"
- WHITE_SPACE: " "
- VALUE_PARAMETER_LIST: "(e: Exception)"
- WHITE_SPACE: " "
- BLOCK: "{
        fun foo(e: Int) { } 
    }"
```
What I was trying to say is that checking if node has parents with these types isn't enough to prove that it is node from catch condition and not body.

Comment (akuleshov7) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt]: I will check prev keyword then

**Pull request merged**